htdocs/categories/class/categorie.class.php

```diff
@@ -488,9 +488,19 @@ function add_type($obj,$type)
 		if ($type == 'customer') $type='societe';
 		if ($type == 'supplier') $type='fournisseur';
 
-		$column_name=$type;
-        if ($type=='contact') $column_name='socpeople';
-        if ($type=='fournisseur') $column_name='societe';
+		/**
+		 * llx_categorie_contact => fk_socpeople
+		 * llx_categorie_fournisseur, llx_categorie_societe => fk_soc
+		 * llx_categorie_member => fk_member
+		 * llx_categorie_product => fk_product
+		 */
+		if ($type == 'contact') {
+			$column_name = 'socpeople';
+		} elseif ($type == 'supplier' || ($type == 'societe')) {
```

Comment:
Yep, 389 has to be removed and a backward compatibility case added like:
```php
// Backward compatibility
if($type == 'fournisseur') $type='supplier'
```
Also make sure no code relies on ```type='fournisseur'```.